### ISSUE TITLE
Article Meta Keywords, Tags, and What's Related

### ISSUE THREAD

**Issue by eSilverStrike** [Geeklog-Core/geeklog#729]
NEW FEATURE DESCRIPTION NOT FINISHED WILL UPDATE

**What's Related**
Appearing with the text "What's Related" as a block below the article

**Related Topics**
Appearing with the text "Filed Under" right under article

**Related Articles By Keyword** 
Appearing with the text "You might also like" under article

**Related Items**
An autotag which returns a list of the newest and recently modified items that are related based on
 the topics passed or that the object belongs too (ie you can return a list of articles and staticpages which relate to your item)

**Comment by mystralkk** [comment]
@eSilverStrike , are you thinking of shipping a new or updated Tags plugin with Geeklog installation?

**Comment by eSilverStrike** [comment]
Honestly I don't know. A part of me wants to keep Geeklog core nice and small but another part of me wants to include plugins that most websites would use. It's also your plugin and you are part of the core Geeklog development team so it is your decision as well. 😊

Since Geeklog is security minded I do think we NEED to include the recaptcha plugin. I've been meaning to bring this up with you to see what you think.

I've also thought about including the menu plugin (as most webmasters want to make deeper menus) and maybe the ban plugin for security reasons. More work may be needed on these 2 though just to tidy up the code and features.

Maybe it's a good time to figure this out since the next version is a major revision 2.2.0

**Comment by mystralkk** [comment]
I agree the reCAPTCHA and Ban plugins (and maybe two-factor authentication) are a must for the Secure CMS.  I also agree modern CMSs will badly need tags and a menu builder.  On the other hand, I don't think the links and calendar plugins are so important.

**Comment by eSilverStrike** [comment]
I agree about dumping the links plugin from the core install as well. Not sure about the Calendar Plugin with organizations/companies that hold events etc.

Another option is If we get the old plugin repository (from Google Summer of Code) up and running where it would be easy for webmasters to find new plugins, find updates to plugins right from within Geeklog. We could then drop all non security essential plugins from core (core developers would still maintain them though) since webmaster could find the plugins easy enough. We could even allow them to pick the plugins from within the install/upgrade of Geeklog as well.

These are just ideas and I am not sure how much work the plugin repository would require.

What do you think we should do? 

1) Look into the Plugin Repository:
(Repository Manager) http://project.geeklog.net/cgi-bin/hgwebdir.cgi/gsoc-2009-rmanager/
(Repository update built into Geeklog) http://project.geeklog.net/cgi-bin/hgwebdir.cgi/gsoc-2009-tpatrick/

2) or Stick with just adding/removing plugins we deem should be in the Geeklog Core Install

**Comment by eSilverStrike** [comment]
Just added Repository to GitHub

https://github.com/Geeklog-GSoC

**Comment by eSilverStrike** [comment]
I am currently looking into the repository and am getting a running version going to see if it is still worth adding to Geeklog. I don't think it automatically notifies the webmaster of updates but it does allow the webmaster to search for them from the plugins page.

You might as well go ahead and add recaptcha. We both agree that is a definite requirement for todays websites.

**Comment by mystralkk** [comment]
I added reCAPTCHA to Geeklog 2.2.0, with support for Invisible reCAPTCHA.
